File: frontend/projects/domain/src/lib/photographer-schedule.ts
import { TimeWindow } from './time-window';
export interface PhotographerSchedule {
  id: string;
  version: number;
  photographerId: string;
  workingWindows: TimeWindow[];
  unavailableWindows: TimeWindow[];
  buffers: {
    before: number;
    after: number;
  };
}

File: backend/src/Qbs.Domain/Entities/PhotographerSchedule.cs
using Qbs.Domain.ValueObjects;

namespace Qbs.Domain.Entities;

public sealed class PhotographerSchedule : Entity
{
    public Guid PhotographerId { get; set; }
    public TimeWindow[] WorkingWindows { get; set; } = [];
    public TimeWindow[] UnavailableWindows { get; set; } = [];
    public TravelBuffers Buffers { get; set; } = new();
}
